modules/obstacleDetector/src/main.cpp

```diff
@@ -0,0 +1,528 @@
+        {
+            if ((int)c[i][2]==id)
+            {
+                center.x+=c[i][0];
+                center.y+=c[i][1];
+                n++;
+            }
+        }
+        center.x/=n;
+        center.y/=n;
+        return sqrt((center.x*center.x)+(center.y*center.y));
+    }
+
+    /****************************************************************/
+    Matrix clusterData(std::vector<LaserMeasurementData> &d, int &nclusters)
+    {
+        cv::RNG rng;
+        vector<cv::Point2d> input=toCvFeatures(d);
+        vector<int> id(input.size());
+        nclusters=cv::partition(input,id,Dist(dist_thresh));
+        Matrix output(input.size(),3);
+        for (int i=0;i<input.size();i++)
+        {
+            output[i][0]=input[i].x;
+            output[i][1]=input[i].y;
+            output[i][2]=id[i];
+            if (color_map.find(id[i])==end(color_map))
+            {
+                cv::Scalar color(rng.uniform(0,255),rng.uniform(0,255),rng.uniform(0,255));
+                color_map[id[i]]=color;
+            }
+        }
+        return output;
+    }
+
+    /***************************************************************/
+    int removeOutliers(Matrix &c, vector<int> &ids)
+    {
+        Vector id=c.getCol(2);
+        int maxid=yarp::math::findMax(id);
+        int nclusters=0;
+        for (int i=0; i<=maxid; i++)
+        {
+            int npoints=std::count(id.begin(),id.end(),i);
+            if (npoints<=min_points)
+            {
+                update(c,i);
+            }
+            else
+            {
+                ids.push_back(i);
+                nclusters++;
+            }
+        }
+        return nclusters;
+    }
+
+    /****************************************************************/
+    void update(Matrix &c, const int &id)
+    {
+        int j=0;
+        while (true)
+        {
+            if ((int)c[j][2]==id)
+            {
+                c.removeRows(j,1);
+            }
+            else
+            {
+                j++;
+            }
+            if (j>=c.rows())
+            {
+                break;
+            }
+        }
+    }
+
+    /****************************************************************/
+    vector<cv::Point2d> toCvFeatures(std::vector<LaserMeasurementData> &data)
+    {
+        vector<cv::Point2d> cvpnts;
+        for (auto &d:data)
+        {
+            double x,y;
+            d.get_cartesian(x,y);
+            if (std::isinf(x) || std::isinf(y) || std::isnan(x) || std::isnan(y))
+            {
+                continue;
+            }
+            cvpnts.push_back(cv::Point2d(x,y));
+        }
+        return cvpnts;
+    }
+
+    /****************************************************************/
+    double getMinDistance(const double &d1, const double &d2, string &s)
+    {
+        if (d1<d2)
+        {
+            s="front-laser";
+            return d1;
+        }
+        else
+        {
+            s="back-laser";
```

Comment:
I would use the term `rear-laser` in lieu of `back-laser`.

Reply:
Along the same line, I would replace `back` with `rear` everywhere.

Reply:
I used `back` to be consistent with the name of the port opened by the laser, which is `/SIM_CER_ROBOT/laser/back:o`.

Reply:
Ok, so the "typo" is upstream 😉 
It's up to you then.

Reply:
I went for `rear`.